plugins/org.jkiss.dbeaver.ext.clickhouse/src/org/jkiss/dbeaver/ext/clickhouse/model/ClickhouseDataSource.java

```diff
@@ -87,10 +90,20 @@ ClickhouseTableEngine getEngineByName(@NotNull String engineName) {
     @Override
     protected Properties getAllConnectionProperties(@NotNull DBRProgressMonitor monitor, JDBCExecutionContext context, String purpose, DBPConnectionConfiguration connectionInfo) throws DBCException {
         Properties properties = super.getAllConnectionProperties(monitor, context, purpose, connectionInfo);
-        if (!isDriverVersionAtLeast(0,8) && !properties.containsKey(ClickhouseConstants.PROP_USE_SERVER_TIME_ZONE)) {
-            properties.put(ClickhouseConstants.PROP_USE_SERVER_TIME_ZONE, "false");
+        if (Objects.equals(properties.getProperty(ClickhouseConstants.PROP_USE_SERVER_TIME_ZONE), "false") && !properties.containsKey(
+            ClickhouseConstants.PROP_USE_TIME_ZONE) || Objects.equals(
+            properties.getProperty(ClickhouseConstants.PROP_USE_TIME_ZONE),
+            "false"
```

Comment:
This feels a bit cramped, don't you think? You can use `CommonUtils#toBoolean` to convert string values to booleans. Moreover, that way, you don't even have to check if the value is present in the map beforehand.

So it's effectively `!PROP_USE_SERVER_TIME_ZONE && !PROP_USE_TIME_ZONE`, right?

```suggestion
        if (!CommonUtils.toBoolean(properties.getProperty(ClickhouseConstants.PROP_USE_SERVER_TIME_ZONE) &&
            !CommonUtils.toBoolean(properties.getProperty(ClickhouseConstants.PROP_USE_TIME_ZONE))
```